modules/core/build.gradle.kts

```diff
@@ -14,8 +16,8 @@ val currentOs = org.gradle.internal.os.OperatingSystem.current()
 
 kotlin {
     jvm {
-        compilations.all {
-            kotlinOptions.jvmTarget = "1.8"
+        compilerOptions {
+            jvmTarget.set(JvmTarget.JVM_21)
```

Comment:
Does this affect which Android platform phoenix is compatible with ?

Reply:
Good point! It seems we can maybe use [desugaring](https://developer.android.com/studio/write/java11-default-support-table). Otherwise we'll be stuck on JDK 8.

cc @dpad85

Reply:
We should keep targeting 1.8, on Phoenix the minimum Android version is API 26 (Android 8).